src/app/modules/simulation/simulation.service.ts

```diff
@@ -174,6 +180,9 @@ export class SimulationService implements OnDestroy, SimulationServiceBase {
             '@id'?: string,
             '@type': string[] | string, // `@type`is certain, because this expandedActions is framed by type searching
           }) => {
+            if (Object.entries(doc).length === 0 && doc.constructor === Object) {
```

Comment:
Why constructor need to be checked? Isn't it sufficient to check the length of entries only?

Reply:
Sorry, I just used [this code](https://stackoverflow.com/questions/679915/how-do-i-test-for-an-empty-javascript-object) as it is. I'll remove `&& doc.constructor === Object.`